lib/ret/hub.ex

```diff
@@ -142,13 +151,46 @@ defmodule Ret.Hub do
     end
   end
 
+  def get_scene_or_scene_listing_by_id(nil) do
+    SceneListing.get_random_default_scene_listing()
+  end
+
+  def get_scene_or_scene_listing_by_id(id) do
+    Scene.scene_or_scene_listing_by_sid(id)
+  end
+
   def get_by_entry_code_string(entry_code_string) when is_binary(entry_code_string) do
     case Integer.parse(entry_code_string) do
       {entry_code, _} -> Hub |> Repo.get_by(entry_code: entry_code)
       _ -> nil
     end
   end
 
+  def get_my_rooms(account, params) do
+    Hub
+    |> where([h], h.created_by_account_id == ^account.account_id and h.entry_mode == ^"allow")
+    |> order_by(desc: :inserted_at)
+    |> preload(^Hub.hub_preloads())
```

Comment:
Is it actually necessary to pin (`^`) `hub_preloads` here? Perhaps it doesn't matter, just curious.

Reply:
I think this is not performing elixir's [pin operator](https://hexdocs.pm/elixir/master/Kernel.SpecialForms.html#%5E/1) but instead the syntax (`^` caret) is borrowed by Ecto's [`preload` macro](https://github.com/elixir-ecto/ecto/blob/99dff4c4403c258ea939fe9bdfb4e339baf05e13/lib/ecto/query.ex#L1990) so that it knows it needs to escape `Hub.hub_preloads()` during macro expansion.

This line starts the escape: https://github.com/elixir-ecto/ecto/blob/99dff4c4403c258ea939fe9bdfb4e339baf05e13/lib/ecto/query/builder/preload.ex#L145

This is where the `^` is consumed: 
https://github.com/elixir-ecto/ecto/blob/99dff4c4403c258ea939fe9bdfb4e339baf05e13/lib/ecto/query/builder/preload.ex#L63-L65

This error results if I don't prepend the `^`: https://github.com/elixir-ecto/ecto/blob/99dff4c4403c258ea939fe9bdfb4e339baf05e13/lib/ecto/query/builder/preload.ex#L67

Reply:
Ah, it didn't occur to me that we were still constructing a query with a macro at this point. TIL! Thanks for digging into the explanation.